Sources/Amplitude/Plugins/iOS/iOSUserInteractionCapture/CustomUIGestureRecognizers.swift

```diff
@@ -0,0 +1,312 @@
+    private static let minimumSlideAmount: Float = 0.01
+
+    /// The delegate that manages user interactions options.
+    private var captureDelegateHandler: () -> UserInteractionCaptureDelegate?
+
+    /// The target element when the screen was first touched.
+    private var initialTarget: AccessibilityTarget?
+
+    /// The value of the target element when the screen was first touched.
+    private var initialValue: Float?
+
+    // MARK: - Life Cycle
+
+    init(for captureDelegateHandler: @escaping () -> UserInteractionCaptureDelegate?) {
+        self.captureDelegateHandler = captureDelegateHandler
+        super.init(target: nil, action: nil)
+    }
+
+    // MARK: - Overridden Methods
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesBegan(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        guard
+            let initialTouchLocation = touches.first?.location(in: captureDelegate.keyWindow)
+        else {
+            return
+        }
+
+        // The target of an interaction is the underlying target behind the
+        // first point when a touch begins since a slider could be modified
+        // even when the final touch point is outside of its bounds.
+        if let target = captureDelegate.accessibilityTargets.first(where: {
+            $0.shape.contains(initialTouchLocation) &&
+            $0.type.contains(.adjustable)
+        }),
+           let slider = target.object as? UISlider
+        {
+            initialTarget = target
+            initialValue = slider.value
+        }
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesEnded(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        defer {
+            self.initialTarget = nil
+            self.initialValue = .zero
+        }
+
+        guard
+            let initialTarget,
+            let initialValue,
+            let slider = initialTarget.object as? UISlider,
+            slider.value != initialValue,
+            abs(initialValue - slider.value) >= GlobalUISlideGestureRecognizer.minimumSlideAmount
+        else {
+            return
+        }
+
+        let percentage = Int(round(slider.value * 100))
+
+        captureDelegate.amplitude?.track(event: UserInteractionEvent(
+            .sliderChanged(to: percentage),
+            label: initialTarget.label,
+            value: initialTarget.value,
+            type: initialTarget.type))
+    }
+}
+
+// MARK: -
+
+internal final class GlobalUIPressGestureRecognizer: UIGestureRecognizer {
+
+    // MARK: - Private Properties
+
+    /// The activation delta between the initial touch point and the final touch point.
+    private static let pressActivationDelta: Double = 10
```

Comment:
This is generally not what the system regards as a "press", please see the documentation for UIControl.Event.touchUpInside.

Reply:
This delta seems to be the minimal difference between the initial and final touch points since anything more might cause a scroll or swipe if the touch was on a scrollable element. Although this might not detect some presses, it avoid false positive where the change in touch positions was for a scroll, but a press is detected.

Reply:
For something like UIButton, the default touchUpInside behavior would detect a touch down, swipe anywhere in bounds, then touch up over any period of time as a single event. I think auto tracking will replace manually firing an event from the default handler (ie, addTarget(_:action:for:) for UIButton), so we should get as close to this behavior as possible.